### PR TITLE
feat: version 2.0

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
BREAKING CHANGE: complete refactor not backward compatible

### Description

Refactoring to extend and implement full TIM automation flows

**Check the relevant boxes:**
- [ ] Bug fix (nonbreaking change that fixes an issue)
- [ ] New feature (nonbreaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [x] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [ ] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

File: examples/bucket-without-tracking-monitoring/variables.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API Token"
```
Comment (ocofaigh) [inline examples/bucket-without-tracking-monitoring/variables.tf:3]: sensitive = true

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,201 @@
+##############################################################################
+# cos-base-module
```
Comment (ocofaigh) [inline main.tf:2]: terraform-ibm-cos

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Tests
+
+For information about how to create and run tests, see the [Tests](https://github.ibm.com/GoldenEye/documentation/blob/master/tests.md) doc.
```
Comment (ocofaigh) [inline tests/README.md:3]: update to same as https://github.com/terraform-ibm-modules/terraform-ibm-module-template/blob/main/tests/README.md

File: .github/settings.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+#
+# When you have installed the GitHub App "repo-settings" in this repository,
+# any change of this settings.yml file is detected by the GitHub App and
+# the settings of this repository are updated immediately.
+#
+_extends: repo-settings
+
+repository:
+  # The name of the repository. Changing this will rename the repository
+  name: terraform-ibm-cos
```
Comment (vburckhardt) [inline .github/settings.yml:12]: description would be aligned with what is currently used internally = "Creates generic COS instance and bucket with a variety of options "

File: README.md [hunk 5]
```diff
@@ -1,180 +1,148 @@
...
-    days    = 0
-    type    = "GLACIER"
-  }
-  expire_rules = [{
-    rule_id = local.expire_rule_id
-    enable  = true
-    days    = 365
-    prefix  = "logs/"
-  }]
+# Replace "master" with a GIT release version to lock into a specific release
```
Comment (vburckhardt) [inline README.md:29]: master -> main

File: README.md [hunk 6]
```diff
@@ -1,180 +1,148 @@
...
+module "cos_module" {
+  source            = "git::https://github.com/terraform-ibm-modules/terraform-ibm-cos?ref=main"
+  environment_name  = "us-staging"
+  resource_group_id = var.resource_group_id
+  ibm_region        = "us-south"
+  cos_key_ring_name = "cos-key-ring"
```
Comment (vburckhardt) [inline README.md:35]: this should be removed as not in current version

File: README.md [hunk 7]
```diff
@@ -1,180 +1,148 @@
...
+  cos_key_name      = ["cos-key"]
```
Comment (vburckhardt) [inline README.md:36]: this should be removed as not in current version

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**